editor/scene_tree_editor.cpp

```diff
@@ -424,7 +424,9 @@ bool SceneTreeEditor::_add_nodes(Node *p_node, TreeItem *p_parent, bool p_scroll
 		item->set_as_cursor(0);
 	}
 
-	bool keep = (filter.is_subsequence_ofn(String(p_node->get_name())));
+	bool is_kept_by_type = filter.begins_with(String("type:")) && filter.replace_first(String("type:"), "").is_subsequence_ofn(String(p_node->get_class()));
```

Comment:
```suggestion
	bool is_kept_by_type = filter.begins_with("type:") && filter.trim_prefix("type:").is_subsequence_ofn(p_node->get_class());
```
* You don't have to explicitly construct `String` from `char const *`.
* You can use `trim_prefix()` to remove prefix :)

Reply:
> * In case you want to change things like the placeholder, it's set here: https://github.com/godotengine/godot/blob/b0ba9468ee40dad636dc9c5cd7b53c24041390d8/editor/scene_tree_dock.cpp#L3361

OK, I'll try it.